packages/flutter_tools/bin/xcode_debug.js

```diff
@@ -529,3 +573,91 @@ function stopApp(xcode, args) {
+ */
+function waitForConfigurationBuildDirToUpdate(targetWorkspace, args) {
+  // Get the project
+  let project;
+  try {
+    project = targetWorkspace.projects().find(x => x.name() == args.projectName);
+  } catch (e) {
+    return new FunctionResult(null, `Failed to find project ${args.projectName}: ${e}`);
+  }
+  if (project == null) {
+    return new FunctionResult(null, `Failed to find project ${args.projectName}.`);
+  }
+
+  // Get the target
+  let target;
+  try {
+    // The target is probably named the same as the project, but if not, just use the first.
+    const targets = project.targets();
+    target = targets.find(x => x.name() == args.projectName);
+    if (target == null && targets.length > 0) {
+      target = targets[0];
+      if (args.verbose) {
+        console.log(`Failed to find target named ${args.projectName}, picking first target: ${target.name()}.`);
+      }
+    }
+  } catch (e) {
+    return new FunctionResult(null, `Failed to find target: ${e}`);
+  }
+  if (target == null) {
+    return new FunctionResult(null, `Failed to find target.`);
+  }
+
+  try {
+    // Use the first build configuration (Debug). Any should do since they all
+    // include Generated.xcconfig.
+    const buildConfig = target.buildConfigurations()[0];
+    const buildSettings = buildConfig.resolvedBuildSettings().reverse();
+
+    // CONFIGURATION_BUILD_DIR is often at (reverse) index 225 for Xcode
+    // projects, so check there first. If it's not there, search the build
+    // settings (which can be a little slow).
+    const defaultIndex = 225;
+    let configurationBuildDirSettings;
+    if (buildSettings[defaultIndex] != null && buildSettings[defaultIndex].name() === 'CONFIGURATION_BUILD_DIR') {
+      configurationBuildDirSettings = buildSettings[defaultIndex];
+    } else {
+      configurationBuildDirSettings = buildSettings.find(x => x.name() === 'CONFIGURATION_BUILD_DIR');
+    }
+
+    if (configurationBuildDirSettings == null) {
+      // This should not happen, even if it's not set by Flutter, there should
+      // always be a resolved build setting for CONFIGURATION_BUILD_DIR.
+      return new FunctionResult(null, `Unable to find CONFIGURATION_BUILD_DIR.`);
+    }
+
+    // Wait up to 2 minutes for the CONFIGURATION_BUILD_DIR to update to the
+    // expected value.
+    const checkFrequencyInSeconds = 0.5;
+    const maxWaitInSeconds = 2 * 60; // 2 minutes
+    const verboseLogInterval = 10 * (1 / checkFrequencyInSeconds);
+    const iterations = maxWaitInSeconds * (1 / checkFrequencyInSeconds);
+    for (let i = 0; i < iterations; i++) {
+      const verbose = args.verbose && i % verboseLogInterval === 0;
+
+      const configurationBuildDir = configurationBuildDirSettings.value();
+      if (verbose) {
+        console.log(`CONFIGURATION_BUILD_DIR: ${configurationBuildDir}`);
+      }
+      if (configurationBuildDir === args.expectedConfigurationBuildDir) {
+        return new FunctionResult(null, null);
+      }
```

Comment:
suggestion, fleshing the logs out a little bit more so at a glance it's clear we're still polling. something like:

```suggestion
      if (configurationBuildDir === args.expectedConfigurationBuildDir) {
        console.log(`CONFIGURATION_BUILD_DIR: ${configurationBuildDir}`);
        return new FunctionResult(null, null);
      }
      if (verbose) {
        console.log(`Current CONFIGURATION_BUILD_DIR: ${configurationBuildDir} while expecting ${args.expectedConfigurationBuildDir}`);
      }
```